src/schema/slack/functions/_scripts/src/templates/template_utils.ts

```diff
@@ -0,0 +1,62 @@
+import { pascalCase } from "https://deno.land/x/case@v2.1.0/mod.ts";
+import { FunctionParameter, FunctionRecord } from "../types.ts";
+import SchemaTypes from "../../../../../schema_types.ts";
+import SlackTypes from "../../../../schema_types.ts";
+import { InternalSlackTypes } from "../../../../types/custom/mod.ts";
+import { AllowedTypeValue, AllowedTypeValueObject } from "./types.ts";
+
+export function autogeneratedComment(): string {
+  const time = new Date();
+  return `/** This file was autogenerated on ${time.toDateString()}. Follow the steps in src/schema/slack/functions/README.md to rebuild **/`;
+}
+
+export function renderFunctionImport(callbackId: string): string {
+  return `import ${getFunctionName(callbackId)} from "./${callbackId}.ts";`;
+}
+
+export function getFunctionName(callbackId: string): string {
+  return pascalCase(callbackId);
+}
+
+export function getSlackCallbackId(
+  defineFunctionInput: FunctionRecord,
+): string {
+  return `slack#/functions/${defineFunctionInput.callback_id}`;
+}
+
+export function getParameterType(type: AllowedTypeValue): string {
+  return type instanceof Object ? type.id : type;
+}
+
+/** @description returns a list of all fn parameters */
+const getParameterList = (fn: FunctionRecord): FunctionParameter[] => [
+  ...fn?.input_parameters ?? [],
+  ...fn?.output_parameters ?? [],
```

Comment:
nit: We don't need the optional chain here do we?? `fn` is always going to be defined

Reply:
Yes, same with `output_parameters` and `input_parameters` because of how I have defined the types